test/conformance/ptp/ptp.go

```diff
@@ -943,3 +989,124 @@ func ptpEventEnabled() bool {
+
+	var masterID string
+	var slaveMasterID string
+	grandMaster := "assuming the grand master role"
+	for _, pod := range ptpRunningPods {
+		if fullConfig.DiscoveredMasterPtpConfig.Label != "" &&
+			podRole(pod, fullConfig.DiscoveredMasterPtpConfig.Label) {
+			podLogs, err := pods.GetLog(&pod, PtpContainerName)
+			Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+			Expect(podLogs).Should(ContainSubstring(grandMaster),
+				fmt.Sprintf("Log message \"%s\" not found in pod's log %s", grandMaster, pod.Name))
+			for _, line := range strings.Split(podLogs, "\n") {
+				if strings.Contains(line, "selected local clock") && strings.Contains(line, "as best master") {
+					// Log example: ptp4l[10731.364]: [eno1] selected local clock 3448ed.fffe.f38e00 as best master
+					masterID = strings.Split(line, " ")[5]
+				}
+			}
+		}
+		if podRole(pod, fullConfig.DiscoveredSlavePtpConfig.Label) {
+			podLogs, err := pods.GetLog(&pod, PtpContainerName)
+			Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+
+			for _, line := range strings.Split(podLogs, "\n") {
+				if strings.Contains(line, "new foreign master") {
+					// Log example: ptp4l[11292.467]: [eno1] port 1: new foreign master 3448ed.fffe.f38e00-1
+					slaveMasterID = strings.Split(line, " ")[7]
+				}
+			}
+		}
+	}
+	Expect(masterID).NotTo(BeNil())
+	Expect(slaveMasterID).NotTo(BeNil())
+	Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
+
+	Expect(masterID).NotTo(BeNil())
+	Expect(slaveMasterID).NotTo(BeNil())
+	Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
+}
+
+func checkRestart(pod v1core.Pod) {
+	fmt.Println("Restarting a node")
+
+	timeout := 5 * time.Minute
+	pollingInterval := 3 * time.Second
+
+	Eventually(func() error {
+		_, err := pods.ExecCommand(client.Client, pod, "container-00", []string{"chroot", "/host", "shutdown", "-r"})
+		return err
+	}, timeout, pollingInterval).Should(BeNil())
+}
+
+func rebootNode(nodesCount int32, fullConfig testconfig.TestConfig) {
+
+	logrus.Info("Rebooting system starts ..............")
+	daemonSetName := "ptp-reboot"
+	namespace := "default"
+	containerName := "container-00"
+	imageWithVersion := "quay.io/testnetworkfunction/debug-partner:latest"
+
+	// 1. create a daemon set for the node reboot
+	timeout := 5 * time.Minute
+	runningPods, err := daemonsets.CreateDaemonSet(daemonSetName, namespace, containerName, imageWithVersion, timeout)
+	if err != nil {
+		fmt.Printf("Error : +%v", err.Error())
+	}
+	// 2. Reboot the machines
+	for _, pod := range runningPods.Items {
+		checkRestart(pod)
+	}
+	// Get all nodes
+	nodes, err := client.Client.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		panic(err.Error())
+	}
+	// Check that all nodes are not unreachable
```

Comment:
// Check that all nodes are not unreachable -> // Wait for all nodes to be unreachable